fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java

```diff
@@ -190,8 +199,16 @@ protected Pair<List<Long>, Long> getSampleTablets() {
             for (int i = 0; i < tabletCounts; i++) {
                 int seekTid = (int) ((i + seek) % ids.size());
                 long tabletId = ids.get(seekTid);
-                sampleTabletIds.add(tabletId);
                 long tabletRows = materializedIndex.getTablet(tabletId).getMinReplicaRowCount(p.getVisibleVersion());
+                if (tabletRows > MAXIMUM_SAMPLE_ROWS) {
+                    // Skip very large tablet.
+                    largeTabletId = tabletId;
```

Comment:
we should record the smallest one in the large tablet list

Reply:
Fixed